pkg/virt-controller/watch/application.go

```diff
@@ -1001,3 +1009,27 @@ func (vca *VirtControllerApp) setupLeaderElector() (err error) {
 
 	return
 }
+
+func (vca *VirtControllerApp) getProductNameAndComponent() (string, string) {
```

Comment:
nit: It would be better to implement it similarly to `shouldChangeRateLimiter` or `configModificationCallback`. Alternatively, we could define args if we think it should be static.

Reply:
maybe should be static, they should be the same as virt-controller itself
so a change in those labels means virt-operator will create a new virt-controller deployment

Reply:
for the args, you mean the args as in:

```
- args:
    - --launcher-image
    - quay.io/openshift-cnv/container-native-virtualization-virt-launcher-rhel9@sha256:8e6582ed31cf7daceeade5d90e307df168ba7723cdaf96c9ca90134b65335d70
    - --exporter-image
    - quay.io/openshift-cnv/container-native-virtualization-virt-exportserver-rhel9@sha256:f48f1cbf4269d1c7a42405ee0fb45379a9bd253cf73828cd7da56704796cf5fc
    - --port
    - "8443"
    - -v
    - "2"
    - -productName
    - XPTO
    - -productComponent
    - XPTO2
    command:
    - virt-controller
```

Reply:
Yes and they are defined [here](https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-controller/watch/application.go#L868)

Reply:
the product name and component labels on `virt-controller` are being reconciled, but the command arguments are not
do you have any idea why?

Reply:
updated